kms/src/main/java/com/example/Snippets.java

```diff
@@ -0,0 +1,448 @@
+    // Create the KeyRing for your project.
+    KeyRing keyring = kms.projects().locations().keyRings()
+        .create(parent, new KeyRing())
+        .setKeyRingId(ringId)
+        .execute();
+
+    System.out.println(keyring);
+    return keyring;
+  }
+
+  /**
+   * Creates a new crypto key with the given id.
+   */
+  public static CryptoKey createCryptoKey(String projectId, String ringId, String keyId)
+      throws IOException {
+    String location = "global";
+    // Create the Cloud KMS client.
+    CloudKMS kms = createAuthorizedClient();
+
+    // The resource name of the location associated with the KeyRing.
+    String parent = String.format(
+        "projects/%s/locations/%s/keyRings/%s", projectId, location, ringId);
+
+    // This will allow the API access to the key for encryption and decryption.
+    String purpose = "ENCRYPT_DECRYPT";
+    CryptoKey cryptoKey = new CryptoKey();
+    cryptoKey.setPurpose(purpose);
+
+    // Create the CryptoKey for your project.
+    CryptoKey createdKey = kms.projects().locations().keyRings().cryptoKeys()
+        .create(parent, cryptoKey)
+        .setCryptoKeyId(keyId)
+        .execute();
+
+    System.out.println(createdKey);
+    return createdKey;
+  }
+
+  /**
+   * Disables the given version of the crypto key.
+   */
+  public static CryptoKeyVersion disableCryptoKeyVersion(
+      String projectId, String ringId, String keyId, String version)
+      throws IOException {
+    String location = "global";
+    // Create the Cloud KMS client.
+    CloudKMS kms = createAuthorizedClient();
+
+    // The resource name of the cryptoKey version
+    String cryptoKeyVersion = String.format(
+        "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s/cryptoKeyVersions/%s",
+        projectId, location, ringId, keyId, version);
+
+    CryptoKeyVersion newVersionState = new CryptoKeyVersion()
+        .setState("DISABLED");
+
+    CryptoKeyVersion response = kms.projects().locations().keyRings().cryptoKeys()
+        .cryptoKeyVersions()
+        .patch(cryptoKeyVersion, newVersionState)
+        .setUpdateMask("state")
+        .execute();
+
+    System.out.println(response);
+    return response;
+  }
+
+  /**
+   * Marks the given version of a crypto key to be destroyed at a scheduled future point.
+   */
+  public static CryptoKeyVersion destroyCryptoKeyVersion(
+      String projectId, String ringId, String keyId, String version)
+      throws IOException {
+    String location = "global";
+    // Create the Cloud KMS client.
+    CloudKMS kms = createAuthorizedClient();
+
+    // The resource name of the cryptoKey version
+    String cryptoKeyVersion = String.format(
+        "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s/cryptoKeyVersions/%s",
+        projectId, location, ringId, keyId, version);
+
+    DestroyCryptoKeyVersionRequest destroyRequest = new DestroyCryptoKeyVersionRequest();
+
+    CryptoKeyVersion destroyed = kms.projects().locations().keyRings().cryptoKeys()
```

Comment:
I kind of like having most of the path in the same line, because it's just boilerplate to get to the main thing you care about (which is on the next line). That way the thing you care about is distinct 
from the noise.